### ISSUE TITLE
Need help to run the tool, no result for sample.mem

### ISSUE THREAD

**Issue by amrita1983** [AlgoLab/galig#4]
Hi,
I am currently working on RNA seq data (Paired end), of human sample, I need to check the alternative splicing event of a specific gene "NEK1" in the sample data, and may be after that for all the genes, my sample has no replicate.I was trying to use ASGAL tool, but while I am giving the input for SpliceAwareAligner tools for generatinig sample.mem file form fastq files nothing is coming up as output. I am using the annotation gtf for that gene only extracted from hg19 human genome gtf.
Here is the command I am usingbin/SpliceAwareAligner -g hg19.fa -a annotation_NEK1.gtf -s NEK1_NM1.fastq -o asgal/NEK1_NM1_output.memPlease help me to run the tool I need the data urgently, can you please help with this?

**Comment by ldenti** [comment]
Hi,
if the process ends without any error and the .mem file is empty, then there could be two reasons behind this behavior:
* none of the input reads align to the splicing graph of that gene, maybe it is necessary to use different values of the parameters -l and -e or maybe most of the reads align to novel exons, that are currently not fully support by ASGAL
* there is some problem in the input data. For example, the format of some input file (such as the GTF) is different from the one expected by ASGAL

If it is possible, can you please share your data with me? It would be easier for me to troubleshoot the problem

**Comment by amrita1983** [comment]
Hi,
Thanks for the reply. I am sharing the data in fastq file reads (for the NEK1 gene sequences that I have extracted from alignment BAM file of the sample) which I  have used in your tool as input for the program SpliceAwareAligner with the following annotation.gtf that I have attached whcih is having the coordinates of NEK1 gene exons, which I have extracted from human hg19. gtf.
command I have used:
/bin/SpliceAwareAligner -g hg19.fa -a annotation_NEK1.gtf -s NEK1_NM1.fastq -o asgal/NEK1_NM1_output.mem

python3 scripts/detectEvents.py -g hg19.fa -a annotation_NEK1.gtf -m asgal/NEK1_NM1_output.mem -o asgal/NEK1_NM1_output.events.csv
I need to check the AS events for this gene NEK1 only for the time being.
It would be a great help if you can test with these files.
Thanks in advance
[NEK1_NM1.fastq.gz](https://github.com/AlgoLab/galig/files/3696577/NEK1_NM1.fastq.gz)

[annotation_NEK1.gtf.gz](https://github.com/AlgoLab/galig/files/3696582/annotation_NEK1.gtf.gz)

**Comment by ldenti** [comment]
Ok, I think I figured out what the problem is. If you directly use the SpliceAwareAligner executable, you should not use the full genome as reference (ie the hg19.fa) but you should use the sequence of the chromosome the gene belongs to. In our case, you should use [this file](https://hgdownload.soe.ucsc.edu/goldenPath/hg19/chromosomes/chr4.fa.gz). I tried with this file and I got a non-empty .mem file ([NEK1_NM1.mem.tar.gz](https://github.com/AlgoLab/galig/files/3697160/NEK1_NM1.mem.tar.gz)).

Let me know if this helps you

**Comment by amrita1983** [comment]
Thanks for the reply.
Sorry ,but I didn't see any output in that NEK1_NM1.mem.tar.gz file that you have attached. 
It has a file called aligns.mem which is empty :(
I tried with the same chr4.fa after indexing it with samtools faidx, and used the same annotation_NEK1.gtf.gz file as gtf (attached earlier) but still that *.mem file is empty for me.
Not getting what is wrong ??

**Comment by ldenti** [comment]
Oh, I'm sorry... I attached the wrong file. [This](https://github.com/AlgoLab/galig/files/3706013/NEK1_NM1.mem.tar.gz) is the correct one.

In any case, it's very strange that you are still getting an empty .mem file. Can you please paste here the exact command you are running?

**Comment by amrita1983** [comment]
Oh, thanks for the correct file. Here is the command that I am using
/bin/SpliceAwareAligner -g chr4.fa -a annotation_NEK1.gtf -s NEK1_NM1.fastq -o asgal/NEK1_NM1_output.mem

I have tried the next command detectEvents.py using the .mem file that you have sent, there also I have got a lot of error messages !! (attaching file)
[error_msg_detectEvents.txt](https://github.com/AlgoLab/galig/files/3706221/error_msg_detectEvents.txt)

python3 scripts/detectEvents.py -g chr4.fa -a annotation_NEK1.gtf -m asgal/NEK1_NM1.mem -o asgal/NEK1_NM1_output.events.csv

I hope I am using all the correct commands.

**Comment by ldenti** [comment]
It's strange that you are still getting an empty .mem file. Does SpliceAwareAligner return any error?

Regarding the detectEvents.py script, if a file `annotation_NEK1.gtf.db` exists, try to remove it and rerun the script.

**Comment by amrita1983** [comment]
No, SpliceAwareAligner did not return any error in nohup also. I have oe more sample to run with same command that you have used for SpliceAwareAligner.

For detectEvents now after removing gtf.db that .csv generated with headers line only.
Here is one more sample file I have to run for the entire set.

[NEK1_AF2.fastq.gz](https://github.com/AlgoLab/galig/files/3707142/NEK1_AF2.fastq.gz)

**Comment by ldenti** [comment]
[This](https://github.com/AlgoLab/galig/files/3707479/NEK1_AF2.mem.tar.gz) is the .mem file for the new sample. If the CSV is empty, it means that no events have been found. Maybe using a lower value for the parameter -l may change something

**Comment by ldenti** [comment]
Anyway, I have no idea why you obtain an empty .mem file.

**Comment by amrita1983** [comment]
Sure I will check to change the parameters, thanks for the help. Such a prompt help is really great help for my work. Please check if you can find some issue and let me know.

**Comment by amrita1983** [comment]
Can you please send me the command that you are using for SpliceAwareAligner, actually I want to see the command as well as the parameters I want ot change -l and -e but I cannot do it in my system still its not giving me any output

**Comment by ldenti** [comment]
The command I used is:
```bash
./bin/SpliceAwareAligner -l 15 -g /data/amrita1983_git/chr4.fa -a /data/amrita1983_git/annotation_NEK1.gtf -s /data/amrita1983_git/NEK1_NM1.fastq -o /data/amrita1983_git/NEK1_NM1.mem
```
